core/src/main/java/org/elasticsearch/plugins/RemovePluginCommand.java

```diff
@@ -81,10 +83,17 @@ void execute(Terminal terminal, String pluginName, Map<String, String> settings)
         }
 
         terminal.println(VERBOSE, "Removing: " + pluginDir);
-        Path tmpPluginDir = env.pluginsFile().resolve(".removing-" + pluginName);
+        final Path tmpPluginDir = env.pluginsFile().resolve(".removing-" + pluginName);
         Files.move(pluginDir, tmpPluginDir, StandardCopyOption.ATOMIC_MOVE);
         pluginPaths.add(tmpPluginDir);
 
         IOUtils.rm(pluginPaths.toArray(new Path[pluginPaths.size()]));
+
+        // we preserve the config files in case the user is upgrading the plugin, but we print
+        // a message so the user knows in case they want to remove manually
+        final Path pluginConfigDir = env.configFile().resolve(pluginName);
+        if (Files.exists(pluginConfigDir)) {
+            terminal.println("-> Preserving plugin config files: " + pluginConfigDir);
```

Comment:
should we say something about upgrades in the message? I suspect people with "uninstall" in their head will have a WTF moment.  Probably good to say "delete manually if not needed".